src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs

```diff
@@ -0,0 +1,36 @@
+#if (NativeAot)
+using System.Text.Json.Serialization;
+#endif
+using Company.ApiApplication1;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Logging.AddConsole();
+
+#if (NativeAot)
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+});
+
+#endif
+var app = builder.Build();
+
+var sampleAlbums = AlbumGenerator.GenerateAlbums().ToArray();
+
+var api = app.MapGroup("/albums");
+api.MapGet("/", () => sampleAlbums);
+api.MapGet("/{id}", (int id) =>
+    sampleAlbums.FirstOrDefault(a => a.Id == id) is { } album
+        ? Results.Ok(album)
+        : Results.NotFound($"Album with id {id} not found"));
+
+app.Run();
+
+#if (NativeAot)
```

Comment:
is there supposed to be a `#endif` for this?

Reply:
Known issue in templating: https://github.com/dotnet/templating/issues/4988